rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -475,6 +495,14 @@ private void refundTxSender(
             generateRejectionRelease(btcTx, btcRefundAddress, rskTx, amount);
         } else {
             logger.debug("[refundTxSender] No btc refund address provided, couldn't get sender address either. Can't refund");
+
+            if (activations.isActive(ConsensusRule.RSKIP181)) {
+                if (activations.isActive(ConsensusRule.RSKIP170)) {
+                    eventLogger.logUnrefundablePegin(btcTx, UnrefundablePeginReason.PEGIN_V1_REFUND_ADDRESS_NOT_SET);
+                } else {
+                    eventLogger.logUnrefundablePegin(btcTx, UnrefundablePeginReason.LEGACY_PEGIN_UNDETERMINED_SENDER);
+                }
```

Comment:
This is not correct.
The status of RSKIP170 doesn't ensure the pegin was a pegin-v1. pegin v1 and pegin legacy will co exist after Iris activation.
You can check the type of pegin using the peginInformation object (see https://github.com/rsksmart/rskj/blob/master/rskj-core/src/main/java/co/rsk/peg/PeginInformation.java#L44 )
```suggestion
                if (peginInformation.getProtocolVersion() == 1) {
                    eventLogger.logUnrefundablePegin(btcTx, UnrefundablePeginReason.PEGIN_V1_REFUND_ADDRESS_NOT_SET);
                } else {
                    eventLogger.logUnrefundablePegin(btcTx, UnrefundablePeginReason.LEGACY_PEGIN_UNDETERMINED_SENDER);
                }
```
This could be improved by at least encoding the `1` in an enum. What do you think @marcos-iov ?

Reply:
Good catch! Yes, an enum with the different peg-in protocol versions (0 and 1 for the moment) should do it

Reply:
thanks @josedahlquist. Applied your suggestion.